onnxscript/rewriter/pattern.py

```diff
@@ -1542,23 +1545,34 @@ def _copy_for_function(
     """Utility function to extract a subgraph out as a function."""
     value_map: dict[ir.Value, ir.Value] = {}
     function_inputs: list[ir.Value] = []
+    constant_nodes: list[ir.Node] = []
     for input in inputs:
         # Create a function input (formal-parameter value) to represent this value:
-        if input is None:
-            raise NotImplementedError("None inputs not supported.")
-        new_value = ir.Value(
-            name=input.name,
-            shape=input.shape,
-            type=input.type,
-            doc_string=input.doc_string,
-        )
+        new_value = (
+            ir.Value(
+                name=input.name,
+                shape=input.shape,
+                type=input.type,
+                doc_string=input.doc_string,
+            )
+            if input
+            else ir.Value()
+        )  # dummy parameter for a None input
         value_map[input] = new_value
         function_inputs.append(new_value)
 
     def copy_value(value: ir.Value | None) -> ir.Value | None:
         if value is None:
             return None
         if value not in value_map:
+            const_value = value.const_value
+            if isinstance(const_value, (ir.Tensor, ir.TensorProtoTensor)):
```

Comment:
```suggestion
            if const_value is not None:
```

Reply:
The const value can be of different types too, but if it is not None it should implement TensorProtocol

Reply:
So, I assume any valid value will be acceptable as parameter to `ir.AttrTensor` down below? Or, is a conversion required in any case?

Reply:
Yes. Any TensorProtocol can be used for `ir.AttrTensor`